### PR TITLE
DDCE-5324 Rewrite to migrate to new ui-test-runner 

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -11,7 +11,9 @@ object Dependencies {
     "io.cucumber"          % "cucumber-junit"    % "7.14.0"   % Test,
     "junit"                % "junit"             % "4.13.2"   % Test,
     "com.novocode"         % "junit-interface"   % "0.11"     % Test,
-    "uk.gov.hmrc"         %% "webdriver-factory" % "0.46.0"   % Test
+    "uk.gov.hmrc"         %% "ui-test-runner"    % "0.17.0"   % Test,
```
Comment (frikit) [inline project/Dependencies.scala:14]: ther eis a lot of cleanup here so I tried to see if I can do and I have this file now feel free to do more cleanup:
```scala
import sbt.*

object Dependencies {

  val test: Seq[ModuleID]    = Seq(
    "com.github.sbt" % "junit-interface" % "0.13.3",
    "io.cucumber"    % "cucumber-junit"  % "7.15.0",
    "io.cucumber"   %% "cucumber-scala"  % "8.20.0",
    "uk.gov.hmrc"   %% "ui-test-runner"  % "0.17.0"
  ).map(_ % Test)

  def apply(): Seq[ModuleID] = test

}


```

File: project/plugins.sbt [hunk 2]
```diff
@@ -6,7 +6,9 @@ resolvers += Resolver.url(
   Resolver.ivyStylePatterns
 )
 
-addSbtPlugin("uk.gov.hmrc"         % "sbt-auto-build" % "3.14.0")
+addSbtPlugin("uk.gov.hmrc"         % "sbt-auto-build" % "3.20.0")
 addSbtPlugin("org.scalameta"       % "sbt-scalafmt"   % "2.4.6")
```
Comment (frikit) [inline project/plugins.sbt:10]: plase update

File: project/plugins.sbt [hunk 3]
```diff
@@ -6,7 +6,9 @@ resolvers += Resolver.url(
...
 addSbtPlugin("com.timushev.sbt"    % "sbt-updates"    % "0.6.3")
```
Comment (frikit) [inline project/plugins.sbt:11]: please update

File: project/plugins.sbt [hunk 4]
```diff
@@ -6,7 +6,9 @@ resolvers += Resolver.url(
...
 addSbtPlugin("com.beautiful-scala" % "sbt-scalastyle" % "1.5.1")
```
Comment (frikit) [inline project/plugins.sbt:12]: please update

**Review by frikit — CHANGES_REQUESTED** [review]
- delete `drivers` folder
- fix mongo version now is 4.0 but we are using 5.0 in README.MD line 17
- add shell or bash to all scripts in readme 
- there is no more tagging for `Tagging tests for ZAP` section in readme we run all tests if there are any specific tags we need to cleanup them
- fix `Configuring the browser to proxy via ZAP` section update new property name and we are not using webdriver-factory
- fix `Executing a ZAP test` section we dont tag tests we run all of them, but we need that script for local usage only
- there is no more `remote-chrome` please change it to `chrome`
- delete section `Installing local driver binaries`
- optimize imports:
```scala
[warn] /Users/victorosipov/IdeaProjects/advance-valuation-rulings-ui-tests/src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/ChangeImporterRoleSteps.scala:20:34: Unused import
```
- please scalastyle warnings
- please delete `start_docker_with_chrome.sh` and add the one from the guide or other ATs we did for ui-test-runner (it has a lot of issues, starting with having chromium v 114 when we support 117 or that image was built only for arm64 memory is off...)
- can be deleted `stop_docker_chrome.sh` as new script will always name container chrome or firefox and you can do docker stop chrome and that is it
- you have a lot of runner what you need its only one to run all tests or 2 having wip runner as well(rest of the scripts and runners)

File: project/Dependencies.scala (hunk 1 above)
Comment (mywyau) [reply project/Dependencies.scala]: thank you for cleaning up the app dep files should be updated now. Not sure if there is anything else to take out tho.

File: project/plugins.sbt (hunk 3 above)
Comment (mywyau) [reply project/plugins.sbt]: should be done

File: project/plugins.sbt (hunk 4 above)
Comment (mywyau) [reply project/plugins.sbt]: should be done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Hooks.scala [hunk 5]
```diff
@@ -16,17 +16,18 @@
...
 
+import io.cucumber.scala.{EN, ScalaDsl}
 import uk.gov.hmrc.test.ui.driver.BrowserDriver
 
-import io.cucumber.scala.{EN, ScalaDsl, Scenario}
-import org.openqa.selenium.{OutputType, TakesScreenshot}
+import uk.gov.hmrc.selenium.webdriver.Browser
 
-class Hooks extends ScalaDsl with EN with BrowserDriver {
-  After { scenario: Scenario =>
```
[the remark this answers is not in the document]
Comment (mywyau) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/Hooks.scala:25]: will add back in forgot to add back in

File: src/test/resources/features/AdminOfOrganisation.feature [hunk 6]
```diff
@@ -1,4 +1,4 @@
-@FunctionalTests @ZAP @EndToEndJourney
```
[the remark this answers is not in the document]
Comment (mywyau) [reply src/test/resources/features/AdminOfOrganisation.feature:1]: okay should be done

File: src/test/resources/features/AgentActingOnBehalfOrganisationAssistant.feature [hunk 7]
```diff
@@ -1,7 +1,7 @@
 @EndToEndJourney
 Feature: Agent Acting on Behalf of Organisation End to End Journey
 
-  @AccessibilityTest @ZAP
```
[the remark this answers is not in the document]
Comment (mywyau) [reply src/test/resources/features/AgentActingOnBehalfOrganisationAssistant.feature:4]: okay should be done

**Comment by mywyau** [comment]
So thinking we should just have the two runners one for all tests, the other for individual tags, tests do take longer to run overall but don't think some tests were ever being ran since only tests tagged with @EndToEndJourney was being ran in the build jobs.

**Review by frikit — APPROVED** [review]

**Pull request merged**